### PR TITLE
PERF: any/all with axis=1

### PR DESCRIPTION
**Pull request by jbrockmendel**
```
from asv_bench.benchmarks.frame_methods import *
self = Dropna()
self.setup("any", 0)

%timeit self.time_dropna_axis_mixed_dtypes("any", 0)
23.6 ms ± 524 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)  # <- master
10.3 ms ± 190 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)  # <- PR
```

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -10349,6 +10349,39 @@ def _logical_func(
             )
             return res._logical_func(name, func, skipna=skipna, **kwargs)
 
+        if (
+            self.ndim > 1
+            and axis == 1
+            and len(self._mgr.arrays) > 1
+            and bool_only is not None
+        ):
+            # Try to avoid a potentially-expensive transpose
```
Comment (jreback) [inline pandas/core/generic.py:10358]: can you throw here on down into a function and call it. (i now it has to be on self) but maybe _reduce_maybe_without_transpose or _reduce_fast or similar

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/generic.py:10358]: sure.  will need to do this to eventually extend this to sum and prod too.

Comment (jbrockmendel) [reply pandas/core/generic.py:10358]: (xref #44793)

**Comment by jbrockmendel** [comment]
updated+greenish (CI failure is codecov)

**Review by jreback — APPROVED** [review]

**Pull request merged**